homework/jenya_s/homework8.py

```diff
@@ -0,0 +1,37 @@
+from collections.abc import Iterable
+
+def bubble_sort(iter_obj, key=None, reverse=False):
+    
+    # first, we check if argument is iterable
+    # if yes and if it is not 'list', we convert the argument to a list
+    
+    if isinstance(iter_obj, Iterable):
+
+        # here we check if some function was passed as a key
+        # and modify our iterable applying the function to all its elements
+        
+        if key == None:
+            iter_obj = list(iter_obj)
+            pass
+        else:
+            iter_obj = list(map(key, iter_obj))
```

Comment:
you shouldn't change items inside the list. 
For example I want sort my list of `{'value': 42}` items, but your sorted function returns list of `42` (with `key=lambda x: x['value']`)